contracts/protocol/libraries/logic/PositionMoverLogic.sol

```diff
@@ -148,4 +157,127 @@ library PositionMoverLogic {
             })
         );
     }
+
+    function executeClaimOtherdeedAndSupply(
+        mapping(address => DataTypes.ReserveData) storage reservesData,
+        mapping(uint256 => address) storage reservesList,
+        DataTypes.UserConfigurationMap storage userConfig,
+        DataTypes.OtherdeedClaimParams memory params
+    ) external returns (uint256) {
+        DataTypes.ReserveData storage reserve = reservesData[params.otherdeed];
+        DataTypes.ReserveCache memory reserveCache = reserve.cache();
+
+        ValidationLogic.validateWithdrawERC721(
+            reservesData,
+            reserveCache,
+            params.otherdeed,
+            params.otherdeedIds
+        );
+
+        DataTypes.TimeLockParams memory timeLockParams;
+
+        (
+            uint64 oldCollateralizedBalance,
+            uint64 newCollateralizedBalance
+        ) = INToken(reserveCache.xTokenAddress).burn(
+                msg.sender,
+                address(this),
+                params.otherdeedIds,
+                timeLockParams
+            );
+
+        bool isWithdrawCollateral = (newCollateralizedBalance <
```

Comment:
I think we don't need this `isWithdrawCollateral`, just need to check if `newCollateralizedBalance == 0`, then we can call `setUsingAsCollateral`